pluginlib/include/pluginlib/class_loader_imp.hpp

```diff
@@ -390,39 +392,77 @@ std::vector<std::string> ClassLoader<T>::getAllLibraryPathsToTry(
 
   const std::string path_separator = getPathSeparator();
 
-  std::vector<std::string> all_paths;
+  std::vector<std::string> all_paths;  // result of all pairs to search
+
   std::string package_prefix = ament_index_cpp::get_package_prefix(exporting_package_name);
-  std::vector<std::string> all_paths_without_extension = {
+
+  // Setup the directories to look in.
+  std::vector<std::string> all_search_paths = {
     // for now just try lib and lib64 (and their respective "libexec" directories)
     package_prefix + path_separator + "lib",
     package_prefix + path_separator + "lib64",
+    package_prefix + path_separator + "bin",  // also look in bin, for dll's on Windows
     package_prefix + path_separator + "lib" + path_separator + exporting_package_name,
     package_prefix + path_separator + "lib64" + path_separator + exporting_package_name,
+    package_prefix + path_separator + "bin" + path_separator + exporting_package_name,
   };
+
+  // Prepare to setup the relative file paths.
   bool debug_library_suffix = (0 == class_loader::systemLibrarySuffix().compare(0, 1, "d"));
   std::string non_debug_suffix;
   if (debug_library_suffix) {
     non_debug_suffix = class_loader::systemLibrarySuffix().substr(1);
   } else {
     non_debug_suffix = class_loader::systemLibrarySuffix();
   }
-  std::string library_name_with_extension = library_name + non_debug_suffix;
   std::string stripped_library_name = stripAllButFileFromPath(library_name);
-  std::string stripped_library_name_with_extension = stripped_library_name + non_debug_suffix;
 
-  for (const auto & current_path : all_paths_without_extension) {
-    all_paths.push_back(current_path + path_separator + library_name_with_extension);
-    all_paths.push_back(current_path + path_separator + stripped_library_name_with_extension);
+  std::string library_name_alternative;  // either lib<library> or <library> without lib prefix
+  const char * lib_prefix = "lib";
+  if (library_name.rfind(lib_prefix, 0) == 0) {
+    library_name_alternative = library_name.substr(strlen(lib_prefix));
+    RCUTILS_LOG_WARN_NAMED("pluginlib.ClassLoader",
```

Comment:
:+1: thanks for adding the warning that'll help converge to the new way of defining them